pkg/apis/core/validation/shoot_test.go

```diff
@@ -5230,6 +5236,58 @@ var _ = Describe("Shoot Validation Tests", func() {
 			)),
 			Entry("at least one worker pool allows system components", zero, zero, one, one, true, true, BeEmpty()),
 		)
+
+		DescribeTable("validate maximum node count",
+			func(kubernetesVersion string, max1, max2 int, allowSystemComponents1, allowSystemComponents2 bool, zones1, zones2 []string, matcher gomegatypes.GomegaMatcher) {
+				workers := []core.Worker{
+					{
+						Name:    "one-active",
+						Minimum: one,
+						Maximum: int32(max1),
+						SystemComponents: &core.WorkerSystemComponents{
+							Allow: allowSystemComponents1,
+						},
+						Zones: zones1,
+					},
+					{
+						Name:    "two-active",
+						Minimum: one,
+						Maximum: int32(max2),
+						SystemComponents: &core.WorkerSystemComponents{
+							Allow: allowSystemComponents2,
+						},
+						Zones: zones2,
+					},
+				}
+
+				Expect(ValidateSystemComponentWorkers(kubernetesVersion, workers, field.NewPath("workers"))).To(matcher)
+			},
+
+			Entry("maximum == len(zones)", "v1.27", three, one, true, false, []string{"1", "2", "3"}, []string{"1"}, BeEmpty()),
+			Entry("maximum < len(zones)", "1.27", two, one, true, false, []string{"1", "2", "3"}, []string{"1"}, ConsistOf(
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeForbidden),
+					"Field": Equal("workers[0].maximum"),
+				})),
+			)),
+			Entry("maximum < len(zones) with multiple system component worker pools in different zones", "1.27", two, one, true, true, []string{"1", "2", "3"}, []string{"3", "4", "5"}, ConsistOf(
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeForbidden),
+					"Field": Equal("workers[0].maximum"),
+				})),
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeForbidden),
+					"Field": Equal("workers[1].maximum"),
+				})),
+			)),
```

Comment:
Can you also add a passing test with two worker pools that covers [this logic](https://github.com/gardener/gardener/pull/7878/commits/aa65e60c5b4831ad2460ff3c2b9929ef14f97c53#diff-51d6dcfca3b6142beb72780e52d0a554bff707dc2aeebe35df9edc6b05aa297fR1779-R1786)?